packages/x-charts-pro/src/Heatmap/Heatmap.tsx

```diff
@@ -203,16 +228,25 @@ const Heatmap = React.forwardRef(function Heatmap(
       onAxisClick={onAxisClick}
       plugins={HEATMAP_PLUGINS}
     >
-      <ChartsSurface ref={ref} sx={sx}>
-        <g clipPath={`url(#${clipPathId})`}>
-          <HeatmapPlot slots={slots} slotProps={slotProps} />
-          <ChartsOverlay loading={loading} slots={slots} slotProps={slotProps} />
-        </g>
-        <ChartsAxis slots={slots} slotProps={slotProps} />
-        <ChartsClipPath id={clipPathId} />
-        {children}
-      </ChartsSurface>
-      {!loading && <Tooltip {...slotProps?.tooltip} />}
+      <ChartsWrapper {...chartsWrapperProps}>
+        {!hideLegend && (
+          <ChartsLegend
+            slots={{ legend: slots?.legend ?? ContinuousColorLegend }}
+            slotProps={{ legend: { labelPosition: 'extremes', ...slotProps?.legend } }}
```

Comment:
Do we want to spread the other slots like we do for other charts?

```suggestion
            slots={{ ...slots, legend: slots?.legend ?? ContinuousColorLegend }}
            slotProps={{ ...slotProps, legend: { labelPosition: 'extremes', ...slotProps?.legend } }}
```